frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceForm.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TextInput, FormGroup, Form } from '@patternfly/react-core';
+import { Identifier } from '~/types';
+import CountFormField from './CountFormField';
+import { validateDefaultCount, validateMinCount } from './utils';
+
+type NodeResourceFormProps = {
+  identifier: Identifier;
+  onUpdate: (data: Identifier) => void;
+};
+
+const NodeResourceForm: React.FC<NodeResourceFormProps> = ({ identifier, onUpdate }) => {
+  const handleFieldUpdate = (field: keyof Identifier, value: unknown) => {
+    onUpdate({ ...identifier, [field]: value });
+  };
+
+  return (
+    <Form>
+      <FormGroup label="Resource label" fieldId="resource-label">
+        <TextInput
+          value={identifier.displayName || ''}
+          onChange={(_, value) => handleFieldUpdate('displayName', value)}
+        />
+      </FormGroup>
+
+      <FormGroup label="Resource identifier" fieldId="resource-identifier">
+        <TextInput
+          value={identifier.identifier || ''}
+          onChange={(_, value) => handleFieldUpdate('identifier', value)}
+        />
+      </FormGroup>
```

Comment:
if this identifier is `cpu` or `memory` then it cannot be edited and this field should be disabled